server/status_test.go

```diff
@@ -118,9 +130,96 @@ func TestStatusJson(t *testing.T) {
 			if err != nil {
 				t.Fatal(err)
 			}
+			t.Log(string(body))
 			if matches, err := regexp.Match(spec.expected, body); !matches || err != nil {
 				t.Errorf("expected match: %t; err nil: %s", matches, err)
 			}
 		}
 	}
 }
+
+// TestStatusGossipJson ensures that the output requrest for the full gossip
+// info contains the required fields.
+func TestStatusGossipJson(t *testing.T) {
+	s := StartTestServer(t)
+	defer s.Stop()
+
+	type prefixedInfo struct {
+		Key string                 `json:"Key"`
+		Val []storage.PrefixConfig `json:"Val"`
+	}
+
+	type rangeDescriptorInfo struct {
+		Key string                `json:"Key"`
+		Val proto.RangeDescriptor `json:"Val"`
+	}
+
+	type keyValueStringPair struct {
+		Key string `json:"Key"`
+		Val string `json:"Val"`
+	}
+
+	type keyValuePair struct {
+		Key string                 `json:"Key"`
+		Val map[string]interface{} `json:"Val"`
+	}
+
+	type infos struct {
+		Infos struct {
+			Accounting  *prefixedInfo        `json:"accounting"`
+			FirstRange  *rangeDescriptorInfo `json:"first-range"`
+			Permissions *prefixedInfo        `json:"permissions"`
+			Zones       *prefixedInfo        `json:"zones"`
+			ClusterID   *keyValueStringPair  `json:"cluster-id"`
+			Node1       *keyValuePair        `json:"node:1"`
+		} `json:"infos"`
+	}
+
+	contentTypes := []string{"application/json", "application/x-protobuf", "text/yaml"}
+	for _, contentType := range contentTypes {
+		req, err := http.NewRequest("GET", "http://"+s.Addr+statusGossipKeyPrefix, nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+		req.Header.Set("Accept", contentType)
+
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != 200 {
+			t.Errorf("Unpexected status code: %v", resp.StatusCode)
+		}
+		contentType = resp.Header.Get(util.ContentTypeHeader)
+		if contentType != "application/json" {
+			t.Errorf("Unpexected content type: %v", contentType)
+		}
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatal(err)
+		}
+		data := &infos{}
+		if err = json.Unmarshal(body, &data); err != nil {
+			t.Fatal(err)
+		}
+		if data.Infos.Accounting == nil {
+			t.Fatalf("no accounting info returned: %v,", body)
```

Comment:
At this point, these should be Errorf instead of Fatalf - that way you can divine more information from a single failure.